nbfetch/handlers.py

```diff
@@ -441,32 +437,56 @@ def get(self):
         self.flush()
 
 
-def _get_user_authentication() -> Tuple[Optional[str], Optional[str]]:
-    """retrieve HS authentication from standard locations(see below) as tuple of username and
-    password. If either cannot be located, both tuple members are be None.
+class UserAccountHandler:
+    """Handler for saving and retrieving user account information
 
-    Note: files take precedence over environment variables
+    Note: files take precedence over environment variables. If the keyfile does not exist, the user
+    and password will be retrieved from environment variables.
     standard file locations:
+        `~/.hs_key`
         `~/.hs_user`
         `~/.hs_pass`
     standard environment variables:
         `HS_USER`
         `HS_PASS`
-
-    Returns
-    -------
-    Tuple[Optional[str], Optional[str]]
-        username, password
     """
+    keyfile = Path("~/.hs_key").expanduser()
     userfile = Path("~/.hs_user").expanduser()
     pwfile = Path("~/.hs_pass").expanduser()
 
-    user = (
-        userfile.read_text().strip()
-        if userfile.exists()
-        else os.getenv("HS_USER", None)
-    )
+    @classmethod
+    def save_account(cls, username: str, password: str) -> None:
+        key = Fernet.generate_key()
+        with open(cls.keyfile, "wb") as f:
+            f.write(key)
+        cypher = Fernet(key)
+        encrypted_user_name = cypher.encrypt(username.encode())
+        encrypted_password = cypher.encrypt(password.encode())
+        with open(cls.userfile, "wb") as f:
+            f.write(encrypted_user_name)
+        with open(cls.pwfile, "wb") as f:
+            f.write(encrypted_password)
+
+    @classmethod
+    def get_account(cls) -> Tuple[Optional[str], Optional[str]]:
+        if cls.keyfile.exists():
```

Comment:
What happens during the "migration" phase as we deploy this out into our GKE images? There will be users who have a `userfile` and `pwfile` but no key

Reply:
In that case the user will be prompted to login to HydroShare. I was able to test that locally by deleting the key file.

Reply:
nice!